test/integration/replicaset/test_replicaset_upgrade.py

```diff
@@ -0,0 +1,245 @@
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+from time import sleep
+
+import pytest
+from replicaset_helpers import stop_application
+from vshard_cluster import VshardCluster
+
+from utils import get_tarantool_version, run_command_and_get_output, wait_file
+
+tarantool_major_version, _ = get_tarantool_version()
+
+
+def run_command_on_instance(tt_cmd, tmpdir, full_inst_name, cmd):
+    con_cmd = [tt_cmd, "connect", full_inst_name, "-f", "-"]
+    instance_process = subprocess.Popen(
+        con_cmd,
+        cwd=tmpdir,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        text=True,
+    )
+    instance_process.stdin.writelines([cmd])
+    instance_process.stdin.close()
+    output = instance_process.stdout.read()
+    return output
+
+
+@pytest.mark.skipif(
+    tarantool_major_version < 3, reason="skip centralized config test for Tarantool < 3"
+)
+def test_upgrade_cluster(tt_cmd, tmpdir_with_cfg):
+    tmpdir = tmpdir_with_cfg
+    app_name = "test_ccluster_app"
+    app_path = os.path.join(tmpdir, app_name)
+    shutil.copytree(os.path.join(os.path.dirname(__file__), app_name), app_path)
+
+    replicasets = [
+        "replicaset-001",
+        "replicaset-002",
+    ]
+
+    try:
+        # Start a cluster.
+        start_cmd = [tt_cmd, "start", app_name]
+        rc, out = run_command_and_get_output(start_cmd, cwd=tmpdir)
+        assert rc == 0
+
+        for i in range(1, 6):
+            file = wait_file(
+                os.path.join(tmpdir, app_name), f"ready-instance-00{i}", []
+            )
+            assert file != ""
+
+        _ = run_command_on_instance(
+            tt_cmd, tmpdir, "test_ccluster_app:instance-004", "box.cfg{read_only=true}"
+        )
+
+        # Instance bootstrap may not have finished yet.
+        sleep(5)
```

Comment:
Please, replace the sleep with some condition. As example, you could try to execute some command on the instance and wait for a bootstrap.

Reply:
I removed this test because it is flaky and duplicates the functionality tested by the last test (`test_upgrade_downgraded_cluster_replicasets`).